src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -43,21 +42,25 @@ internal sealed class TupleTypeSymbol : NamedTypeSymbol, ITupleTypeSymbol
                 out this._fields);
         }
 
+        /// <summary>
+        /// Copy the original tuple, but modify it to use the new underlying type.
+        /// </summary>
         private TupleTypeSymbol(TupleTypeSymbol originalTuple,
                                 NamedTypeSymbol newUnderlyingType)
         {
             _underlyingType = newUnderlyingType;
-            _fields = originalTuple._fields.SelectAsArray(field =>
-                                                          {
-                                                              var newUnderlyingField = field.UnderlyingTypleFieldSymbol.OriginalDefinition.AsMember(_underlyingType);
-                                                              return new TupleFieldSymbol(field.Name, this, newUnderlyingField.Type, newUnderlyingField);
-                                                          });
+
+            _fields = originalTuple._fields.SelectAsArray(field => field?.WithUnderlyingType(this, newUnderlyingType));
         }
 
-        internal static TupleTypeSymbol ConstructTupleTypeSymbol(TupleTypeSymbol originalTupleType, NamedTypeSymbol newUnderlyingType)
+        /// <summary>
+        /// Copy this tuple, but modify it to use the new underlying type.
+        /// </summary>
+        internal TupleTypeSymbol WithUnderlyingType(NamedTypeSymbol newUnderlyingType)
```

Comment:
Yes it is, in `AbstractTypeMap.cs` at line 56. My understanding is this is during type substitution.